app/graphql/mutations/digital_object/create_digital_object.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Mutations
+  module DigitalObject
+    class CreateDigitalObject < Mutations::BaseMutation
+      argument :project, Inputs::Project::StringKey, "String key for project", required: true
+      argument :digital_object_type, Enums::DigitalObjectTypeEnum, "digital object type", required: true
+      argument :descriptive_metadata, GraphQL::Types::JSON, required: true
+      argument :identifiers, [String], required: true
+
+      field :digital_object, Types::DigitalObjectInterface, null: false
+      field :errors, [Types::Errors::FieldedInput], null: false
+
+      def resolve(project:, digital_object_type:, descriptive_metadata:, identifiers:)
+        project = Project.find_by!(project.to_h)
+        ability.authorize! :create_objects, project
+
+        digital_object = Hyacinth::Config.digital_object_types.key_to_class(digital_object_type).new
+        digital_object.primary_project = project
+        digital_object.assign_attributes(
+          'identifiers' => Set.new(identifiers), 'descriptive_metadata' => descriptive_metadata
```

Comment:
The assign_attributes method accepts identifiers as an array and converts them into a set internally (in internal assign_identifiers method), so no need to create a new Set here.

Reply:
Ah, this is an artifact from when I was assigning identifiers separately.